controllers/storagecluster/cephblockpools.go

```diff
@@ -88,10 +88,30 @@ func (o *ocsCephBlockPools) reconcileCephBlockPool(r *StorageClusterReconciler,
 	}
 
 	_, err = ctrl.CreateOrUpdate(r.ctx, r.Client, cephBlockPool, func() error {
-		cephBlockPool.Spec.PoolSpec.DeviceClass = storageCluster.Status.DefaultCephDeviceClass
-		cephBlockPool.Spec.PoolSpec.EnableCrushUpdates = true
-		cephBlockPool.Spec.PoolSpec.FailureDomain = getFailureDomain(storageCluster)
-		cephBlockPool.Spec.PoolSpec.Replicated = generateCephReplicatedSpec(storageCluster, "data")
+		// Pass the poolSpec from the storageCluster CR
+		// Set default values in the poolSpec if necessary
+		cephBlockPool.Spec.PoolSpec = storageCluster.Spec.ManagedResources.CephBlockPools.PoolSpec
+
+		poolSpec := &cephBlockPool.Spec.PoolSpec
+		if poolSpec.DeviceClass == "" {
+			poolSpec.DeviceClass = storageCluster.Status.DefaultCephDeviceClass
+		}
+		poolSpec.EnableCrushUpdates = true
```

Comment:
Any reason to disallow them from overriding this setting? I can't think of a user scenario, but perhaps support could run into some condition where they want to override it.

Reply:
Couldn't do with these booleans as there is no way to tell if a boolean field is either just unset or explicitly set to false. If this was a bool ptr we could have done the check.

Reply:
oh right, nevermind, it doesn't help even if the parent spec is nullable